### ISSUE TITLE
`type[T] -> type[T]` cannot be used in union if assigned to an attribute in a namespace

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1987]
**Describe the bug**
The return type of `type[T] -> type[T]` cannot be used in a union if assigned to an attribute in a namespace.  The snippet below shows it happening with a class but it also happens with modules.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
No errors would be produced.

**Screenshots or Code**
```python
# pyright: strict

from typing import TypeVar, Union

T = TypeVar("T")
TT = TypeVar("TT", bound=type)

def foo1(value: type[T]) -> type[T]:
    ...

def foo2(value: TT) -> TT:
    ...

class bar:
    class baz:
        pass

    @foo1
    class qux:
        pass

    @foo2
    class quux:
        pass

# Ok
baz_or_str: "bar.baz | str"

# Operator "|" not supported for types "Type[qux]" and "Type[str]"
#   Operator "|" not supported for types "Type[qux]" and "Type[str]"
qux_or_str: "bar.qux | str"

# Expected class type but received "Type[qux] | Type[str]"
#   "Type[qux]" is not a class
qux_or_str_Union: Union[bar.qux, str]

# Ok
quux_or_str: "bar.quux | str"

```

**VS Code extension or command-line**
1.1.149

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.150, which I just published. It will also be included in the next release of pylance.
